### PR TITLE
Datumizing Bloodsucker Clans

### LINKED ISSUE #746: Clan-less bloodsuckers can make favorite vassals
More of a self-reminder. Making a favorite vassal without a clan makes the favorite not get any benefits when you do join a clan, ruining the whole point of favorite vassals.

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

I got a shitty laptop to do some work while I'm not home for a week.
I originally made this PR to make sure everything was working, but I got inspiration to expand this PR as much as I did because I had a dream where I coded myself datumizing Bloodsucker clans. I started my work by just copy+pasting the work I did in my dream, then continued from there.

Closes https://github.com/fulpstation/fulpstation/issues/746

Stuff to do:
- [X] The Antag UI should allow them to join a Clan whenever they want, rather than being tied to the Coffin.
- [X] Antag UI should show the description of their Clan.
- [x] Add Clan Icons for each clan
- [x] Make the Antag UI show the Clan's icon in then
- [ ] Testing (I can't test vassal stuff on this shitty laptop help)
- [x] Move Power explanations from Mentor tab to the Antag UI

Changes (that I can remember at the time of writing):
All Bloodsucker clans are now datums, and are mostly handled by signals.
You can't spend any ranks until you join a Clan.
You can join a clan whenever you want using your antag UI
There is no such thing as 'clanless' Bloodsucker anymore
Brujah lost all functionality and is the "new" clanless bloodsucker, as they get no special abilities and is the default Bloodsucker gameplay. I did this because Brujah is already the lamest clan and I don't want there to have Bloodsuckers who don't participate in the Clan system in some way.
This means all Brujah-unique content (Frenzy ability & Drink blood while in Frenzy objective) has been removed.
Fixes Vassals getting two deconversion messages to themselves.
Fixes Vassals saying 'In 3' instead of 'In 30 seconds' when Sol is near.

Removes the Mentor tab's Bloodsucker help button, it's now been replaced.

Screenshots of it in-game:
image
image
image
image
image
image

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
I am going to turn this into a massive out of scope PR that I will never be able to realistically finish.

**Comment by OneAsianTortoise** [comment]
> I am going to turn this into a massive out of scope PR that I will never be able to realistically finish.

the real solution lies in here: d3c85d1

File: tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx [hunk 1]
```diff
@@ -105,22 +152,114 @@ const BloodsuckerIntro = () => {
...
-            selected={tab === 1}
-            onClick={() => setTab(1)}>
-            Introduction
-          </Tabs.Tab>
-        </Tabs>
-        {tab === 1 && <BloodsuckerIntro />}
-      </Window.Content>
-    </Window>
+    <Stack vertical fill>
+      <Stack.Item minHeight="20rem">
```
Comment (raffclar) [inline tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx:182]: A single item stack seems a bit odd.

Comment (raffclar) [reply tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx:182]: I would remove the excess stack and have the Section as the root component

File: tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx [hunk 2]
```diff
@@ -105,22 +152,114 @@ const BloodsuckerIntro = () => {
...
+        <PowerSection />
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const PowerSection = (props: any, context: any) => {
+  const { act, data } = useBackend<BloodsuckerInformation>(context);
+  const { power } = data;
+  const [selectedPower, setSelectedPower] = useSharedState(
```
Comment (raffclar) [inline tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx:218]: Is this interface being used by other clients? `useSharedState` ensures other clients can read/write this state. `useLocalState` may be more appropriate here.

File: tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx [hunk 3]
```diff
@@ -105,22 +152,114 @@ const BloodsuckerIntro = () => {
...
+              height="12rem"
+              as="img"
+              src={resolveAsset(`bloodsucker.${selectedPower.power_icon}.png`)}
+            />
+          )}
+          <Divider Vertical />
+        </Stack.Item>
+        <Stack.Divider />
+        <Stack.Item scrollable grow={1} fontSize="16px">
+          {!!selectedPower && selectedPower.power_explanation}
```
Comment (raffclar) [inline tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx:260]: Boolean true isn't needed. Expression can be boolean true or truthy
```suggestion
          {selectedPower && selectedPower.power_explanation}
```

File: tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx [hunk 4]
```diff
@@ -105,22 +152,114 @@ const BloodsuckerIntro = () => {
...
+            selected={selectedPower.power_name}
+            width="100%"
+            options={power.map((powers) => powers.power_name)}
+            onSelected={(powerName: string) =>
+              setSelectedPower(
+                power.find((p) => p.power_name === powerName) || power[0]
+              )
+            }
+          />
+          {!!selectedPower && (
```
Comment (raffclar) [inline tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx:248]:
```suggestion
          {selectedPower && (
```

File: tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx [hunk 5]
```diff
@@ -105,22 +152,114 @@ const BloodsuckerIntro = () => {
...
+        <Button
+          icon="info"
+          tooltipPosition="left"
+          tooltip={'Select a Power to explain.'}
+        />
+      }>
+      <Stack>
+        <Stack.Item grow>
+          <Dropdown
+            displayText={selectedPower.power_name}
```
Comment (raffclar) [inline tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx:238]:
```suggestion
```

Comment (raffclar) [reply tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx:238]: Isn't needed.

Comment (raffclar) [reply tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx:238]: Dropdown code uses selected anyway `{displayText ? displayText : this.state.selected}`

File: tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx [hunk 6]
```diff
@@ -105,22 +152,114 @@ const BloodsuckerIntro = () => {
...
+            selected={selectedPower.power_name}
+            width="100%"
+            options={power.map((powers) => powers.power_name)}
+            onSelected={(powerName: string) =>
+              setSelectedPower(
+                power.find((p) => p.power_name === powerName) || power[0]
```
Comment (raffclar) [inline tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx:244]:
```suggestion
                power.find((p) => p.power_name === powerName)
```
You shouldn't need an or expression. Something isn't quite right. I have a feeling the `powerName` is undefined.

File: tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx [hunk 7]
```diff
@@ -105,22 +152,114 @@ const BloodsuckerIntro = () => {
...
+  );
+};
+
+const PowerSection = (props: any, context: any) => {
+  const { act, data } = useBackend<BloodsuckerInformation>(context);
+  const { power } = data;
+  const [selectedPower, setSelectedPower] = useSharedState(
+    context,
+    'power',
+    power[1]
```
Comment (raffclar) [inline tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx:221]: I think by defaulting to `power[1]` it's causing the dropdown to fail as it's overflowing. Maybe try 0?
```suggestion
    power[0]
```

File: tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx (hunk 1 above)
Comment (JohnFulpWillard) [reply tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx:182]: When I do this, going to this section of the UI causes the whole UI to scroll down, making the tabs unreachable.

File: tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx (hunk 5 above)
Comment (JohnFulpWillard) [reply tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx:238]: Tried this and it seemed to have broke the entire UI

File: tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx (hunk 6 above)
Comment (JohnFulpWillard) [reply tgui/packages/fulpui-patches/interfaces/AntagInfoBloodsucker.tsx:244]: It worked but vscsode said there was an error, so I kept it but brought the other suggestions anyway.

**Pull request merged**